plugin-core/src/main/java/appland/webviews/navie/NavieEditor.java

```diff
@@ -242,6 +255,22 @@ public void run(@NotNull ProgressIndicator indicator) {
         }.queue();
     }
 
+    private void handleChooseFilesToPin() {
+        DataContext dataContext = dataId -> {
```

Comment:
Instead of the providing a custom data context and emulating how actions are called, I think it would be better to extract a static method in ChooseAndAddNavieContextFilesAction, because its signature would show the needed input data. Providing input via the generic data context is prone to errors.

if a data context was used, it should be created with our DataContexts because of compatibility with the recent SDKs.